src/core/AudioEngine/AudioEngine.cpp

```diff
@@ -494,61 +494,38 @@ AudioOutput* AudioEngine::createDriver( const QString& sDriver )
 
 	if ( sDriver == "OSS" ) {
 		pDriver = new OssDriver( m_AudioProcessCallback );
-		if ( pDriver->class_name() == NullDriver::_class_name() ) {
-			delete pDriver;
-			pDriver = nullptr;
-		}
 	} else if ( sDriver == "JACK" ) {
 		pDriver = new JackAudioDriver( m_AudioProcessCallback );
-		if ( pDriver->class_name() == NullDriver::_class_name() ) {
-			delete pDriver;
-			pDriver = nullptr;
-		} else {
 #ifdef H2CORE_HAVE_JACK
+		if ( dynamic_cast<NullDriver*>(pDriver) == nullptr ) {
 			static_cast<JackAudioDriver*>(pDriver)->setConnectDefaults(
 						Preferences::get_instance()->m_bJackConnectDefaults
 						);
```

Comment:
Would seem to be more natural to write this as:
```c++
if ( auto pJackDriver = dynamic_cast<JackAudioDriver*>( pDriver ) ) {
    pJackDriver->setConnectDefaults(
        Preferences::get_instance()->m_bJackConnectDefaults
    );
}
```
since the conditional code is more conditional on it being an actual `JackAudioDriver*` than just being non-null?